apps/ledger-live-desktop/src/renderer/components/TransactionConfirm/index.tsx

```diff
@@ -137,7 +171,81 @@ const TransactionConfirm = ({ t, device, account, parentAccount, transaction, st
   });
   const key = ("mode" in transaction && transaction.mode) || "send";
   const recipientWording = t(`TransactionConfirm.recipientWording.${key}`);
-  return (
+
+  const typeTransaction: string = (
+    fields.find(
+      (field: { label: string }) => field.label && field.label === "Type",
+    ) as DeviceTransactionField & { value: string }
+  ).value;
+
+  const amountTransaction: string = (
+    fields.find(
+      (field: { label: string }) => field.label && field.label === "Amount",
+    ) as DeviceTransactionField & { value: string }
+  ).value;
+  return typeTransaction === "Approve" ? (
+    <Container>
+      <Container paddingX={26}>
+        <DeviceBlocker />
+        <Animation animation={getDeviceAnimation(device.modelId, type, "verify")} />
+        <Text ff={"Inter|Medium"} textAlign={"center"} fontSize={22} marginBottom={12}>
+          {t("approve.description")}
+        </Text>
+        <Alert type="primary" mb={26}>
+          <Trans
+            i18nKey={
+              amountTransaction === "Unlimited MATIC" ? "approve.unlimited" : "approve.limited"
+            }
+            values={{
+              recipientWording,
+            }}
+          />
+        </Alert>
+        <Box
+          style={{
+            width: "100%",
+          }}
+          mb={20}
+        >
+          {fields.map((field, i) => {
+            const MaybeComponent = fieldComponents[field.type];
+            if (!MaybeComponent) {
+              console.log(
+                `TransactionConfirm field ${field.type} is not implemented! add a generic implementation in components/TransactionConfirm.js or inside families/*/TransactionConfirmFields.js`,
```

Comment:
Does it come from you this `console.log` or was it present before ?

Reply:
Not from me

Reply:
Ok I suppose we can keep this in for now